core/pubkeyConverter/bech32PubkeyConverter.go

```diff
@@ -80,42 +77,39 @@ func (bpc *bech32PubkeyConverter) Decode(humanReadable string) ([]byte, error) {
 }
 
 // Encode converts the provided bytes in a bech32 form
-func (bpc *bech32PubkeyConverter) Encode(pkBytes []byte) string {
+func (bpc *bech32PubkeyConverter) Encode(pkBytes []byte) (string, error) {
 	if len(pkBytes) != bpc.len {
-		bpc.log.Debug("bech32PubkeyConverter.Encode PkBytesLength",
-			"hex buff", hex.EncodeToString(pkBytes),
-			"error", ErrWrongSize,
-			"stack trace", string(debug.Stack()),
-		)
-
-		return ""
+		return "", fmt.Errorf("%w when encoding address, expected length %d, received %d",
+			ErrWrongSize, bpc.len, len(pkBytes))
 	}
 
 	//since the errors generated here are usually because of a bad config, they will be treated here
 	conv, err := bech32.ConvertBits(pkBytes, bech32Config.fromBits, bech32Config.toBits, bech32Config.pad)
 	if err != nil {
-		bpc.log.Warn("bech32PubkeyConverter.Encode ConvertBits",
-			"hex buff", hex.EncodeToString(pkBytes),
-			"error", err,
-			"stack trace", string(debug.Stack()),
-		)
-
-		return ""
+		return "", fmt.Errorf("%w when converting bits", ErrConvertBits)
 	}
 
-	converted, err := bech32.Encode(bech32Config.prefix, conv)
+	converted, err := bech32.Encode(bpc.prefix, conv)
 	if err != nil {
-		bpc.log.Warn("bech32PubkeyConverter.Encode Encode",
-			"hex buff", hex.EncodeToString(pkBytes),
-			"conv", hex.EncodeToString(conv),
-			"error", err,
-			"stack trace", string(debug.Stack()),
-		)
-
-		return ""
+		return "", fmt.Errorf("%w when encoding address", ErrConvertBits)
+	}
+
+	return converted, nil
+}
+
+// EncodeSlice will encode the provided bytes slice
```

Comment:
```suggestion
// EncodeSlice converts the provided bytes slice into a slice of bech32 addresses
```
as to keep same format as for `Encode`